services/venice-router/src/main/java/com/linkedin/venice/router/api/path/VeniceMultiKeyPath.java

```diff
@@ -107,25 +108,7 @@ public void initialize(
           e.getMessage());
     }
 
-    for (ByteBuffer key: keys) {
-      RouterKey routerKey = new RouterKey(key);
-
-      keyNum++;
-
-      if (stats != null) {
-        stats.getStatsByType(RequestType.MULTI_GET).recordKeySize(storeName, routerKey.getKeySize());
-      }
-
-      // partition lookup
-      int partitionId = partitioner.getPartitionId(routerKey.getKeyBuffer(), partitionNum);
-      routerKey.setPartitionId(partitionId);
-      K routerRequestKey = createRouterRequestKey(key, keyIdx, partitionId);
-      this.routerKeyMap.put(routerKey, routerRequestKey);
-      ++keyIdx;
-    }
-    setPartitionKeys(this.routerKeyMap.keySet());
-
-    int keyCount = getPartitionKeys().size();
+    int keyCount = keys.size();
     if (keyCount > maxKeyCount) {
```

Comment:
Curious, how does the request get pass the client key count check ?

Reply:
I think that check is only implemented in (recent versions of) the FC and not in the TC? Also, the check is based on some kind of periodically refreshed metadata that could be out of aync with the backend?

Reply:
Right, I see the check only in FC but not in TC. Should we add the check in TC as well ? it will save lots of resources for router and client as well for large requests.

Reply:
At first glance I would rather not... it seems to be "not very thin" 😅 ... but we can consider it.